### PR TITLE
Add a Prettier task

### PR DESCRIPTION
**Pull request by iamlacroix**
* Adds a `prettier` task to our react-scripts.
* Adds docs for setting up Prettier in an app.
* Moves the @trunkclub/build migration guide into a new /docs folder

Please also review the Prettier setup doc: https://github.com/trunkclub/tcweb-build/blob/add-prettier/docs/PRETTIER.md

### CODE REVIEW

File: docs/PRETTIER.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+It might beneficial to run Prettier once on the app ahead of time,
+and then committing the formatted changes.
+
+This will ensure all diffs going forward are cleaner and easier to understand.
+Otherwise, diffs will have a lot of noise the first time a changed file is reformatted.
+
+This can be by running the following from your project root:
+
+```bash
+node_modules/.bin/tcweb-build prettier "src/**/*.{js,es6}"
```
Comment (sloanelybutsurely) [inline docs/PRETTIER.md]: If someone is using `yarn` (which I think everyone is now) they should be able to issue 

```bash
yarn tcweb-build prettier "src/**/*.{js,es6}
```

Comment (sloanelybutsurely) [reply docs/PRETTIER.md]: Also, should we included `.jsx` in the command? I think we have some of those floating around still.

File: docs/PRETTIER.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
++    "tcweb-build prettier",
++    "git add"
++  ]
++},
+ "dependencies": {},
+```
+
+### Usage
+
+With the above configuration, Prettier will now be run on any staged source files before a commit is completed.
```
Comment (sloanelybutsurely) [inline docs/PRETTIER.md]: Are these hooks required? Should they be listed as something optional?

File: packages/react-scripts/bin/react-scripts.js [hunk 3]
```diff
@@ -11,6 +11,7 @@
 'use strict';
 
 var spawn = require('cross-spawn');
+var path = require('path');
```
Comment (sloanelybutsurely) [inline packages/react-scripts/bin/react-scripts.js]: Is this needed? I don't see where in what you've added that you're using `path`.

**Review by sloanelybutsurely — COMMENTED** [review]
This is great!

I know there are some formatting choices that prettier makes that our eslint configuration does not agree with. If someone gets into one of these situations and has a precommit hook, they will get stuck with a broken build.

Should we include changes to the eslint configurations to ensure this doesn't happen?

File: docs/PRETTIER.md (hunk 1 above)
Comment (iamlacroix) [reply docs/PRETTIER.md]: @zperrault Ah, I did not know we have jsx files. I'll add that.

File: packages/react-scripts/bin/react-scripts.js (hunk 3 above)
Comment (iamlacroix) [reply packages/react-scripts/bin/react-scripts.js]: Good catch.

**Review by eanplatter — APPROVED** [review]

**Review by sloanelybutsurely — APPROVED** [review]

**Comment by sloanelybutsurely** [comment]
Sounds great! Is it worth upgrading to 1.4.0 before merging?

**Pull request merged**